src/web/middleware/extract-user-id-from-headers-middleware.ts

```diff
@@ -0,0 +1,20 @@
+import type { NextFunction, Request, Response } from 'express';
+
+import { UnauthorizedError } from './errors';
+
+import { ensureString } from '../../common/stringUtils';
+
+export const extractUserIdFromHeadersMiddleware = (
```

Comment:
I originally had this middleware also attempt to extract the user ID from the `userId` query parameter, but decided it was better to just do this in the `/checkAuth` route instead (which is the only place where we expect a `userId` in query params).